components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/finality/v1/UtxoReceiveFinalityFlowV1.kt

```diff
@@ -73,17 +78,30 @@ class UtxoReceiveFinalityFlowV1(
             session.send(payload)
             throw CordaRuntimeException(payload.message)
         }
-        transaction = receiveSignaturesAndAddToTransaction(transaction)
-        verifyAllReceivedSignatures(transaction)
-        persistenceService.persist(transaction, TransactionStatus.UNVERIFIED)
+
+        if (waitForAdditionalSignatures) {
+            transaction = receiveSignaturesAndAddToTransaction(transaction)
+            verifyAllReceivedSignatures(transaction)
+            persistenceService.persist(transaction, TransactionStatus.UNVERIFIED)
+        }
```

Comment:
```suggestion
        if (waitForAdditionalSignatures) {
            transaction = receiveSignaturesAndAddToTransaction(transaction)
        }
        verifyAllReceivedSignatures(transaction)
        if (waitForAdditionalSignatures) {
            persistenceService.persist(transaction, TransactionStatus.UNVERIFIED)
        }
```
The `verifyAllReceivedSignatures` is still needed, but the other two are not.